src/infra/azure/storage/queue/signature-request.ts

```diff
@@ -0,0 +1,23 @@
+import { QueueClient } from "@azure/storage-queue";
+
+import * as TE from "fp-ts/TaskEither";
+import { pipe } from "fp-ts/function";
+
+import { SignatureRequest } from "../../../../signature-request/signature-request";
+import { enqueueMessage } from "./client";
+
+export type EnqueueSignatureRequest = (
+  request: SignatureRequest
+) => TE.TaskEither<Error, SignatureRequest>;
+
+export const enqueueSignatureRequest =
```

Comment:
since there's nothing specific to azure here, I guess this file should be inside the use-case directory (the business logic goes there), isn't it @lucacavallaro ?

Reply:
It is here because it depends on the `QueueClient` which is relative to `azure`

Reply:
I see. It looks like the only thing this method add to `enqueueMessage` is the `JSON.stringify `part, probably you don't need this method at all as is. 

What you really want is to define a new type for each of the different kind of messages and an encoder (JSON.stringify) for them. In this case, it would be a method that takes a `signatureRequest` object and converts it to the structure the queue message is expecting. This would enable the consumer and producer to share this type to encode / decode it.

Something like: `SignatureRequestMessage.encode(signatureRequest)`